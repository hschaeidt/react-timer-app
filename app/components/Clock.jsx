import React from 'react';

export default class Clock extends React.Component {
  static formatSeconds(totalSeconds) {
    let seconds = `${totalSeconds % 60}`;
    let minutes = `${Math.floor(totalSeconds / 60)}`;

    if (seconds < 10) {
      seconds = `0${seconds}`;
    }

    if (minutes < 10) {
      minutes = `0${minutes}`;
    }

    return `${minutes}:${seconds}`;
  }

  render() {
    const { totalSeconds } = this.props;

    return (
      <div className="clock">
        <span className="clock-text">
          {Clock.formatSeconds(totalSeconds)}
        </span>
      </div>
    );
  }
}

Clock.propTypes = {
  totalSeconds: React.PropTypes.number,
};

Clock.defaultProps = {
  totalSeconds: 0,
};
