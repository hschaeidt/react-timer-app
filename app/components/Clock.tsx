import * as React from 'react';

interface ClockProps extends React.Props<any> {
  totalSeconds: number,
}

export default class Clock extends React.Component<ClockProps, void> {
  public static defaultProps: ClockProps = {
    totalSeconds: 0,
  }

  public static formatSeconds(totalSeconds: number): string {
    const seconds = totalSeconds % 60;
    const minutes = Math.floor(totalSeconds / 60);

    let strSeconds = `${seconds}`;
    let strMinutes = `${minutes}`;

    if (seconds < 10) {
      strSeconds = `0${seconds}`;
    }

    if (minutes < 10) {
      strMinutes = `0${minutes}`;
    }

    return `${strMinutes}:${strSeconds}`;
  }

  render(): JSX.Element {
    const { totalSeconds } = this.props;

    return (
      <div className="clock">
        <span className="clock-text">
          {Clock.formatSeconds(totalSeconds)}
        </span>
      </div>
    );
  }
}
