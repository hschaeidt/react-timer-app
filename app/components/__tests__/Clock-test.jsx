import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Clock from '../Clock';

describe('<Clock />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Clock totalSeconds={30} />);
    expect(tree).toMatchSnapshot();
  });

  describe('render', () => {
    it('should render clock to output', () => {
      const clock = shallow(<Clock totalSeconds={62} />);
      const actualText = clock.find('.clock-text').text();

      expect(actualText);
    });
  });

  describe('formatSeconds', () => {
    it('should format seconds', () => {
      const seconds = 615;
      const expected = '10:15';
      const actual = Clock.formatSeconds(seconds);

      expect(actual).toBe(expected);
    });

    it('should format seconds when min/sec less than 10', () => {
      const seconds = 61;
      const expected = '01:01';
      const actual = Clock.formatSeconds(seconds);

      expect(actual).toBe(expected);
    });
  });
});
