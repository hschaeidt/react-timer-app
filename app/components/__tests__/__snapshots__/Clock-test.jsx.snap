exports[`<Clock /> renders correctly 1`] = `
<div
  className="clock">
  <span
    className="clock-text">
    00:30
  </span>
</div>
`;
